.github/workflows/dotnet-desktop.yml

```diff
@@ -14,10 +14,73 @@ jobs:
       - name: Set up .NET SDK
         uses: actions/setup-dotnet@v3
         with:
-          dotnet-version: "7.x"
-
+          dotnet-version: "6.0.x" # should be updated to latest .NET version
+      - name: Cache NuGet packages
+        uses: actions/cache@v3
+        with:
+          path: ~/.nuget/packages
+          key: ${{ runner.os }}-nuget-${{ hashFiles('**/*.csproj') }}
```

Comment:
Try include additional files for props and targets in the hash as well

`key: ${{ runner.os }}-nuget-${{ hashFiles('**/*.csproj', '**/*.props', '**/*.targets') }}`